src/EnergyPlus/DaylightingManager.cc

```diff
@@ -1326,266 +1265,186 @@ void CalcDayltgCoeffsMapPoints(EnergyPlusData &state, int const ZoneNum)
 
     if (state.dataDaylightingManager->mapFirstTime && state.dataDaylightingData->TotIllumMaps > 0) {
         IL = -999;
-        for (MapNum = 1; MapNum <= state.dataDaylightingData->TotIllumMaps; ++MapNum) {
+        for (int MapNum = 1; MapNum <= state.dataDaylightingData->TotIllumMaps; ++MapNum) {
             IL = max(IL, state.dataDaylightingData->IllumMapCalc(MapNum).TotalMapRefPoints);
         }
         state.dataDaylightingManager->MapErrIndex.dimension(IL, state.dataSurface->TotSurfaces, 0);
         state.dataDaylightingManager->mapFirstTime = false;
     }
 
-    // Azimuth of view vector in absolute coord sys
-    AZVIEW = (state.dataDaylightingData->ZoneDaylight(ZoneNum).ViewAzimuthForGlare + state.dataHeatBal->Zone(ZoneNum).RelNorth +
-              state.dataHeatBal->BuildingAzimuth + state.dataHeatBal->BuildingRotationAppendixG) *
-             DataGlobalConstants::DegToRadians;
+    int enclNum = state.dataDaylightingData->IllumMapCalc(mapNum).enclIndex;
+    auto &thisEnclDaylight = state.dataDaylightingData->enclDaylight(enclNum);
+
+    // Azimuth of view vector in absolute coord sys - set to zero here, because glare isn't calculated for map points
+    // TODO MJW: remove all of the glare calcs for map points
+    AZVIEW = 0.0;
     // View vector components in absolute coord sys
     VIEWVC(1) = std::sin(AZVIEW);
     VIEWVC(2) = std::cos(AZVIEW);
     VIEWVC(3) = 0.0;
 
-    for (MapNum = 1; MapNum <= state.dataDaylightingData->TotIllumMaps; ++MapNum) {
+    numRefPts = state.dataDaylightingData->IllumMapCalc(mapNum).TotalMapRefPoints;
 
-        if (state.dataDaylightingData->IllumMapCalc(MapNum).Zone != ZoneNum) continue;
+    state.dataDaylightingData->IllumMapCalc(mapNum).DaylIllumAtMapPt = 0.0;  // Daylight illuminance at reference points (lux)
+    state.dataDaylightingData->IllumMapCalc(mapNum).GlareIndexAtMapPt = 0.0; // Glare index at reference points
+    state.dataDaylightingData->IllumMapCalc(mapNum).SolidAngAtMapPt = 0.0;
+    state.dataDaylightingData->IllumMapCalc(mapNum).SolidAngAtMapPtWtd = 0.0;
+    state.dataDaylightingData->IllumMapCalc(mapNum).IllumFromWinAtMapPt = 0.0;
+    state.dataDaylightingData->IllumMapCalc(mapNum).BackLumFromWinAtMapPt = 0.0;
+    state.dataDaylightingData->IllumMapCalc(mapNum).SourceLumFromWinAtMapPt = 0.0;
```

Comment:
Is this called every time step? Why is the entire array zeroed out if we are not doing detailed integration and only the hour subarray zeroed out if we are? Zeroing out 4D arrays is kinda expensive, should try to minimize how much of it we do.